test/lib/ansible_test/_data/pytest/plugins/ansible_pytest_unit_test_module_mocks.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017–2021 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""Pytest fixtures for mocking Ansible modules."""
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from io import BytesIO
+import json
+import sys
+
+import pytest
+
+import ansible.module_utils.basic
+from ansible.module_utils.common._collections_compat import MutableMapping
+from ansible.module_utils.six import PY3, string_types
+from ansible.module_utils._text import to_bytes
+
+
+@pytest.fixture
+def ansible_module_args(mocker, monkeypatch, request):
```

Comment:
_Self note:_ Would it be better to expose a callable for the module? It would basically accept the task args directly.

Something long the lines of
```python
# ---------- global (in a plugin or conftest)
import pytest


@pytest.fixture
def ansible_module_factory(monkeypatch):
    def make_module(module_name):
        def run_with_args(**kwargs):
            monkeypatch.... # the same as in `ansible_module_args`
            return AnsibleModule(module_name...).run(**kwargs)  # maybe convert error into exception (AnsibleModuleError?) and keep success a dict? object?
        return run_with_args
    return make_module


# ---------- local (in a separate test module, this is how the end-users would use it)
import pytest


@pytest.fixture
def call_module_x(ansible_module_factory):  # or should this helper also be global? with a redefinable fixture returning a name?
    return ansible_module_factory('module_x')

def test_module_X(call_module_x):
    """Test something true about module X."""
    assert call_module_x(x='y', u=9) == {'x': 'y', 'u': 9}

def test_module_X_negative(call_module_x):
    """Test error handling in module X."""
    with pytest.raises(AnsibleModuleError, match=r'Not enough args') as module_error:
        call_module_x()

    assert 'x' in module_error.result['missing_args']
```

An alternative API for initializing a module caller could be:
```python
@pytest.fixture
def ansible_module_name():
    return 'module_X'
```
This one seems more ergonomic and native to pytest.

Reply:
_Self note:_ How about having an extra fixture for a subprocess-based module runner (for isolation)?